docs/plugins.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Plugins
+description: Plugins provide powerful ways to extend and customize MyST
+---
+
+Plugins provide powerful ways to extend and customize MyST by adding directives and roles or specific transformations on the AST that can download or augment your articles and documentation. These plugins also support custom rendering and transformation pipelines for various export targets including React, HTML, LaTeX, and Microsoft Word.
+
+:::{danger} Plugins are incomplete in Beta
+The interfaces and packaging for the plugins may change substantially in the future.\
+**Expect changes!!**
+
+If you are implementing a plugin, please let us know on [GitHub](https://github.com/executablebooks/mystmd) or [Discord](https://discord.mystmd.org/).
+:::
+
+## Overview of a Plugin
+
+Plugins are executable javascript code that can modify a document source. The supported plugin types are:
+
+directives
+: Add or overwrite directives, which provide a "block-level" extension point.
+: For example, create a `:::{proof}` extension that allows for numbered proofs.
+
+roles
+: Add or overwrite roles, which provide an inline extension point.
+: For example, create a role for showing units, `` {si}`4 kg per meter squared` ``.
+
+:::{warning} Planned - Not Yet Implemented
+:class: dropdown
+transforms
+: These plugins transform the document source while it is rendered.
+: For example, add metadata or transform a link to an DOI.
+
+renderers
+: These plugins add handlers for various nodes when exporting to a specific format.
+: For example, do something special for node in HTML, React, Microsoft Word, or LaTeX.
+:::
+
+## Creating a Plugin
+
+To create a plugin, you will need a single Javascript file[^esm] that exports one or more of the objects above. For example, a simple directive that pulls a random image from [Unsplash](https://unsplash.com/) can be created with a single file that exports an `unsplash` directive.
+
+[^esm]: The format of the Javascript should be an ECMAScript modules, not CommonJS. This means it uses `import` statements rather than `require()` and is the most modern style of Javascript.
+
+:::{literalinclude} unsplash.mjs
+:filename: unsplash.mjs
+:caption: A plugin to add an `unsplash` directive that includes a beautiful, random picture based on a query string.
+:::
```

Comment:
@nthiery actually using `literalinclude` -- very much enjoying it. :)

This PR is also prompted a bit from your latex requirements, hoping that having access to custom plugins will give you better stop-gap control over your documents while we also figure out the defaults!